domain/model/service/service.go

```diff
@@ -300,20 +296,6 @@ func (s *Service) createModel(
 		)
 	}
 
-	agentVersion := args.AgentVersion
-	if args.AgentVersion == version.Zero {
-		agentVersion = agentVersionSelector()
-	}
-
-	if err := validateAgentVersion(agentVersion, s.agentBinaryFinder); err != nil {
-		return nil, fmt.Errorf(
-			"creating model %q with agent version %q: %w",
-			args.Name, agentVersion, err,
-		)
-	}
-
-	args.AgentVersion = agentVersion
```

Comment:
I'm assuming this is safe to do, because we already know this before hand. This should **always** be the same as the controller, there is no reason to create a model that isn't the same.

Reply:
Correct.